docs/CONTRIBUTING.md

```diff
@@ -14,13 +14,26 @@ If you are a *git ninja*, we feel happy and sad for you at the same time. Your s
 ## Reviews
 Reviews are a great way to get familiar with the codebase. If you would like to start with reviewing just use **[this handy github filter](https://github.com/xbmc/xbmc/pulls?q=is%3Apr+is%3Aopen+review%3Anone)** to see PRs that are in need of one (we have plenty!).
 
-## Pull request guidelines
+## Pull request making guidelines
 * **Create topic branches**. Never ever open a pull request from your master branch. **Ever!**
 * **Code against master branch**. Unless release specific, all Kodi development occurs in the `master` branch. If it needs backporting it can be done after it hits master.
 * **One topic branch per change**. Found something that needs fixing but is unrelated to the current work? Create a new topic branch and open another *pull request*.
 * **Use a coherent commit history**. Make sure each individual commit in your *pull request* is meaningful and organized in logical chunks. Tidy up and squash commits before submitting.
-* **Be descriptive, but concise**. The *pull request* and commits should have a useful title and follow the `[component(s)] Short description...` format. No, `[README.md] Update` is **not** descriptive enough. The *pull request* description should only contain information relevant to the change.
-* **Be patient, be kind, rewind**. We're all volunteers with busy lives but usually very responsive in cataloging *pull requests* and pinging the relevant maintainer(s). Sometimes though, things go unnoticed for longer than usual. Probably due to beer deprivation... or lack thereof. Feel free to nudge us if we forget.
+* **Be descriptive, but concise**. The *pull request* and *commits* should have an useful title and follow the format:
+`[Component(s)] Short description...` or alternative format `Component(s): Short description...`<br/>
+by component we mean for example, the name of the modified class, the code section context (e.g. name of the parent folder), dependency name, etc... You can get hints by browsing the git history of the modified file, an example:<br/>
+`[Python][xbmcgui] Fix x problem...` or `Python/xbmcgui: Fix x problem...`<br/>
+instead something like `[README.md] Update` is **not** descriptive enough.<br/>
+The *pull request* description should only contain information relevant to the change.
+* **Draft PR**. If your work requires further development, and you wish to ask the opinion of other developers on how to proceed, you can publish your PR as Draft.
+
+## Pull request merge guidelines
+* **First rule, be patient, be kind, rewind**. We're all volunteers with busy lives but usually very responsive in cataloging *pull requests* and pinging the relevant maintainer(s).
+* **Jenkins code formatting requests**. Always apply code style formatting requests from Jenkins. If not, please provide a valid reason why you do not apply them.
```

Comment:
We should probably consider to provide a tutorial on how to integrate that with git so that they are automatically applied before pushing. Not asking you to do it :) It's more in the context of our git tutorial rather than here

Reply:
yeah not detailed example here or become heavy to read